internal/pkg/logs/service.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package logs
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/aws/copilot-cli/internal/pkg/aws/cloudwatchlogs"
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+)
+
+const (
+	logGroupNamePattern = "/copilot/%s-%s-%s"
+)
+
+type logGetter interface {
+	TaskLogEvents(logGroupName string,
+		streamLastEventTime map[string]int64,
+		opts ...cloudwatchlogs.GetLogEventsOpts) (*cloudwatchlogs.LogEventsOutput, error)
+}
+
+// ServiceLogs wraps service that writes service log events to a writer.
+type ServiceLogs struct {
+	logGroupName string
+	eventsGetter logGetter
+	w            io.Writer
+}
+
+// WriteLogEventsOpts wraps the parameters to call WriteLogEvents.
+type WriteLogEventsOpts struct {
+	Follow             bool
+	OutputLogs         func(w io.Writer, logs []*cloudwatchlogs.Event) error
+	GetLogEventConfigs []cloudwatchlogs.GetLogEventsOpts
```

Comment:
Right now we're leaking our dependency to the caller, instead can have multiple options in this struct? and then internally we create `[]cloudwatchlogs.GetLogEventsOpts`

Reply:
good call!